File: core/src/commonMain/kotlin/work/socialhub/kmastodon/api/FavouritesResource.java
package work.socialhub.kmastodon.api;

import mastodon4j.Range;
import mastodon4j.entity.Status;
import mastodon4j.entity.share.Response;

/**
 * @author hecateball
 */
public interface FavouritesResource {

    /**
     * Fetching a user's favourites.
     *
     * @return an array of Statuses favourited by the authenticated user
     */
    public Response<Status[]> getFavourites();

    /**
     * Fetching a user's favourites.
     *
     * @return an array of Statuses favourited by the authenticated user
     */
    public Response<Status[]> getFavourites(Range range);
}

File: core/src/commonMain/kotlin/work/socialhub/kmastodon/api/FavouritesResource.kt
package work.socialhub.kmastodon.api

import work.socialhub.kmastodon.api.request.favourites.FavouritesFavouritesRequest
import work.socialhub.kmastodon.api.response.Response
import work.socialhub.kmastodon.api.response.favourites.FavouritesFavouritesResponse
import kotlin.js.JsExport

@JsExport
interface FavouritesResource {

    /**
     * Fetching a user's favourites.
     */
    suspend fun favourites(
        request: FavouritesFavouritesRequest
    ): Response<Array<FavouritesFavouritesResponse>>

    @JsExport.Ignore
    fun favouritesBlocking(
        request: FavouritesFavouritesRequest
    ): Response<Array<FavouritesFavouritesResponse>>
}
